src/ansiblelint/file_utils.py

```diff
@@ -199,12 +200,57 @@ def get(self, key: Any, default: Any = None) -> Any:
 
     @property
     def content(self) -> str:
-        """Retried file content, from internal cache or disk."""
+        """Retrieve file content, from internal cache or disk."""
         if self._content is None:
             with open(self.path.resolve(), mode="r", encoding="utf-8") as f:
                 self._content = f.read()
+            if self._original_content is None:
+                self._original_content = self._content
         return self._content
 
+    @content.setter
+    def content(self, value: str) -> None:
+        """Update ``content`` and calculate ``updated``."""
+        if not isinstance(value, str):
+            raise TypeError(f"Expected str but got {type(value)}")
+        if self._original_content is None:
+            if self._content is not None:
+                self._original_content = self._content
+            elif not self.path.exists():
+                # new file
+                self._original_content = ""
+        self.updated = self._original_content != value
```

Comment:
Hmm. In `content.setter`, should we try to read the file to populate `_original_content` if the file exists and it has not been read yet?

```suggestion
        if self._original_content is None:
            if self._content is not None:
                self._original_content = self._content
            elif self.path.exists():
                # use self.content to populate self._original_content
                _ = self.content
            else:
                # new file
                self._original_content = ""
        self.updated = self._original_content != value
```